heartbeat/LVM-activate

```diff
@@ -738,6 +774,60 @@ tagging_deactivate() {
 	return $OCF_SUCCESS
 }
 
+# Gets a list of device-mapper devices corresponding for $VG/$LV if
+# $LV is defined. Otherwise gets list corresponding to $VG.
+get_dm_devs() {
+	if [ -n "$LV" ]; then
+		select="vg_name=$VG && lv_name=$LV"
+	else
+		select="vg_name=$VG"
+	fi
+
+	dmsetup info --noheadings --noflush -c -S "$select" \
+		| grep -v "^No devices found" \
+		| cut -d":" -f1
+}
+
+# Gets a list of device-mapper dependencies corresponding to the
+# specified device.
+get_dm_dependencies() {
+	dmsetup deps -o blkdevname "$1" \
+		| sed -e "s/.*:[[:blank:]]*\(.*\)/\1/" -e "s/, /,/g" \
+		| tr " " "\n"
+}
+
+# Perform a conditional I/O test based on the value of OCF_CHECK_LEVEL
+deep_status() {
+	case "$OCF_CHECK_LEVEL" in
+		0)
+			return $OCF_SUCCESS
+			;;
+		10)
+			if [ -n "$LV" ]; then
+				dm_name="/dev/$VG/$LV"
+			else
+				# If there are many LVs in the VG dir,
+				# pick the first one.
+				dm_name="/dev/$VG/$(ls -1 /dev/$VG | head -n 1)"
```

Comment:
This needs to be modified for when `partial_activation=true`. In its current state, this is a bug introduced by https://github.com/ClusterLabs/resource-agents/commit/b325f97e. `partial_activation=true` means we should succeed as long as at least one PV is readable. It doesn't have to be the first PV.